packages/wonder-blocks-form/src/components/text-area.tsx

```diff
@@ -0,0 +1,401 @@
+            validate,
+            onValidate,
+            required,
+            resizeType,
+            light,
+            // Should only include aria related props
+            ...otherProps
+        } = props;
+
+        const [error, setError] = React.useState<string | null>(null);
+
+        const ids = useUniqueIdWithMock("text-area");
+        const uniqueId = id ?? ids.get("id");
+
+        const handleChange = (
+            event: React.ChangeEvent<HTMLTextAreaElement>,
+        ) => {
+            const newValue = event.target.value;
+            onChange(newValue);
+            handleValidation(newValue);
+        };
+
+        const handleValidation = (newValue: string) => {
+            if (validate) {
+                const error = validate(newValue) || null;
+                setError(error);
+                if (onValidate) {
+                    onValidate(error);
+                }
+            } else if (required) {
+                const requiredString =
+                    typeof required === "string"
+                        ? required
+                        : defaultErrorMessage;
+                const error = newValue ? null : requiredString;
+                setError(error);
+                if (onValidate) {
+                    onValidate(error);
+                }
+            }
+        };
+
+        useOnMountEffect(() => {
+            // Only validate on mount if the value is not empty. This is so that fields
+            // don't render an error when they are initially empty
+            if (value !== "") {
+                handleValidation(value);
+            }
+        });
+
+        const getStyles = (): (CSSProperties | Falsy)[] => {
+            // Base styles are the styles that apply regardless of light mode
+            const baseStyles = [
+                styles.textarea,
+                typographyStyles.LabelMedium,
+                resizeType && resizeStyles[resizeType],
+            ];
+            const defaultStyles = [
+                styles.default,
+                !disabled && styles.defaultFocus,
+                disabled && styles.disabled,
+                !!error && styles.error,
+            ];
+            const lightStyles = [
+                styles.light,
+                !disabled && styles.lightFocus,
+                disabled && styles.lightDisabled,
+                !!error && styles.lightError,
+            ];
+            return [...baseStyles, ...(light ? lightStyles : defaultStyles)];
+        };
+        return (
+            <View style={{width: "100%"}}>
+                <StyledTextArea
+                    id={uniqueId}
+                    data-testid={testId}
+                    ref={ref}
+                    className={className}
+                    style={[...getStyles(), style]}
+                    value={value}
+                    onChange={handleChange}
+                    placeholder={placeholder}
+                    disabled={disabled}
```

Comment:
**question:** I wonder if you explored using `aria-disabled` here in case we want to allow SRs focus on the field? I'm not sure still if this would be the right approach, but interested in hearing from you as we use it in multiple WB components already.

Reply:
Great question! I was wondering this as well. Since the TextArea component is similar to the TextField component, I followed the behaviour of the TextField component, which was to use the `disabled` attribute directly. As a general guideline, do we want all interactive elements in the design system to utilize `aria-disabled` instead of `disabled` so that they remain focusable?

Reply:
That's right, I think we should try to use that approach as much as we can.

Reply:
Sounds good! I created a follow up task to make the TextArea focusable when it is `disabled` (WB-1734). I'll work on this after our deprecated component migration work this focus week!